### PR TITLE
0.1.2 dev QR support

### PR DESCRIPTION
**Pull request by michilumin**
Know that likely msvc cmake stuff has to be resolved before this goes in

### CODE REVIEW

**Comment by xanimo** [comment]
So a fair bit to do but just for purpose of reminding, perhaps you can drop [4c96f3e](https://github.com/dogecoinfoundation/libdogecoin/pull/94/commits/4c96f3e0ab25ca048f65fb2261284bd094577f1c) and [4992e0b](https://github.com/dogecoinfoundation/libdogecoin/pull/94/commits/4992e0b8cbe24d95f9e7a758e3eceb9adb9958c8) and rebase on top of `0.1.2-dev`? I have 2-3 commits I can open on top to make it pass CI but would need those last 2 commits dropped first:

https://github.com/xanimo/libdogecoin/commits/0.1.2-dev-michi-qr

File: include/dogecoin/qrengine.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+void printQr(const uint8_t qrcode[]);
+int outputQRStringFromQRBytes(const uint8_t* inQrBytes, char* outString);
+int stringToQrArray(const char* inString, uint8_t* outQrBytes);
+
+LIBDOGECOIN_BEGIN_DECL
+
+LIBDOGECOIN_API int qrgen_p2pkh_to_qr_string(const char* in_p2pkh, char* outString);
+LIBDOGECOIN_API void qrgen_p2pkh_consoleprint_to_qr(char* in_p2pkh);
+LIBDOGECOIN_API int qrgen_p2pkh_to_qrbits(const char* in_p2pkh, uint8_t* outQrByteArray);
+LIBDOGECOIN_API int qrgen_string_to_qr_pngfile(const char* outFilename, const char* inString, uint8_t SizeMultiplier);
```
Comment (nooperation) [inline include/dogecoin/qrengine.h:22]: A couple inconsistent parameter names and formatting between declaration and definition

qrengine.h + libdogecoin.h vs qrengine.c:59
```
int qrgen_p2pkh_to_qr_string(const char* in_p2pkh,char* outqr)
int qrgen_p2pkh_to_qr_string(const char* in_p2pkh, char* outString);
```

qrengine.h + libdogecoin.h vs qrengine.c:67
```
int qrgen_p2pkh_to_qrbits(const char* in_p2pkh,uint8_t* qrbits)
int qrgen_p2pkh_to_qrbits(const char* in_p2pkh, uint8_t* outQrByteArray);
```

qrengine.h + libdogecoin.h vs qrengine.c:140
```
int qrgen_string_to_qr_pngfile(const char *filename, const char* inString, uint8_t multiplier)
int qrgen_string_to_qr_pngfile(const char* outFilename, const char* inString, uint8_t SizeMultiplier);
```

File: include/dogecoin/qrengine.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+#ifndef __LIBDOGECOIN_QRENGINE_H__
+#define __LIBDOGECOIN_QRENGINE_H__
+
+#include <dogecoin/cstr.h>
+#include <dogecoin/dogecoin.h>
+
+// internal funcs not exported
+void printQr(const uint8_t qrcode[]);
+int outputQRStringFromQRBytes(const uint8_t* inQrBytes, char* outString);
+int stringToQrArray(const char* inString, uint8_t* outQrBytes);
```
Comment (nooperation) [inline include/dogecoin/qrengine.h:15]: A couple inconsistent parameter names and formatting between declaration and definition

qrengine.h:14 vs qrengine.c:35
```
int outputQRStringFromQRBytes(const uint8_t* inQrBytes, char* outString);
int outputQRStringFromQRBytes(const uint8_t qrcode[],char* outstring)
```

qrengine.h vs qrengine.c:18
```
int stringToQrArray(const char* inString, uint8_t* outQrBytes);
int stringToQrArray(const char* inString,uint8_t* qrcode)
```

File: include/dogecoin/qrengine.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+LIBDOGECOIN_BEGIN_DECL
+
+LIBDOGECOIN_API int qrgen_p2pkh_to_qr_string(const char* in_p2pkh, char* outString);
+LIBDOGECOIN_API void qrgen_p2pkh_consoleprint_to_qr(char* in_p2pkh);
```
Comment (nooperation) [inline include/dogecoin/qrengine.h:20]: Can this be `const char* in_p2pkh`?

**Review by nooperation — COMMENTED** [review]
Just some cosmetic issues

File: include/dogecoin/qrengine.h (hunk 1 above)
Comment (michilumin) [reply include/dogecoin/qrengine.h:22]: Fixed, normalized to what was in header files.

File: include/dogecoin/qrengine.h (hunk 2 above)
Comment (michilumin) [reply include/dogecoin/qrengine.h:15]: Fixed, normalized to what was in qrengine.h

File: include/dogecoin/qrengine.h (hunk 3 above)
Comment (michilumin) [reply include/dogecoin/qrengine.h:20]: Yes, fixed, works.

**Comment by michilumin** [comment]
Latest commit 2680882 should fix all consistency issues in this PR as well as a qrcode-sized memleak (extraneous malloc and unfreed calloc) exactly the size of a qr code.

**Review by edtubbs — APPROVED** [review]
ACK, reviewed and tested on linux, wsl, and msvc.

**Review by xanimo — APPROVED** [review]
ACK

**Pull request merged**